packages/nodejs/package.json

```
@@ -46,13 +48,14 @@
         "tabWidth": 4
     },
     "scripts": {
-        "generate-js": "yarn run grpc_tools_node_protoc --js_out=import_style=commonjs,binary:grpc --grpc_out=grpc_js:grpc --plugin=protoc-gen-grpc=../../node_modules/.bin/grpc_tools_node_protoc_plugin -I ../../deps/concordium-grpc-api ../../deps/concordium-grpc-api/*.proto",
-        "generate-ts": "yarn run grpc_tools_node_protoc --plugin=protoc-gen-ts=../../node_modules/.bin/protoc-gen-ts --ts_out=grpc_js:grpc -I ../../deps/concordium-grpc-api ../../deps/concordium-grpc-api/*.proto",
-        "generate": "([ -e \"../../deps/concordium-grpc-api\" ] && mkdir -p grpc && yarn generate-js && yarn generate-ts) || echo 'Please checkout submodules before building'; false",
+        "generate-js": "yarn run grpc_tools_node_protoc --js_out=import_style=commonjs,binary:grpc --grpc_out=grpc_js:grpc --plugin=protoc-gen-grpc=../../node_modules/.bin/grpc_tools_node_protoc_plugin -I ../../deps/concordium-grpc-api ../../deps/concordium-grpc-api/**/*.proto",
+        "generate-ts": "yarn run grpc_tools_node_protoc --plugin=protoc-gen-ts=../../node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts --ts_out=grpc_js:grpc -I ../../deps/concordium-grpc-api ../../deps/concordium-grpc-api/**/*.proto",
+        "generate-ts-v2": "yarn run grpc_tools_node_protoc --plugin=protoc-gen-ts=../../node_modules/@protobuf-ts/plugin/bin/protoc-gen-ts --ts_out=src/grpc -I ../../deps/concordium-grpc-api ../../deps/concordium-grpc-api/**/*.proto",
```

```suggestion
        "generate-ts-v2": "yarn run grpc_tools_node_protoc --plugin=protoc-gen-ts=../../node_modules/@protobuf-ts/plugin/bin/protoc-gen-ts  --ts_opt optimize_code_size --ts_out=src/grpc -I ../../deps/concordium-grpc-api ../../deps/concordium-grpc-api/**/*.proto",
```

We should check the speed difference, but this helps with the size a lot, removing a 1mb from the unpacked size :thinking:

---

And this shouldn't generate files for the v1 files

---

```suggestion
        "generate-ts-v2": "yarn run grpc_tools_node_protoc --plugin=protoc-gen-ts=../../node_modules/@protobuf-ts/plugin/bin/protoc-gen-ts --ts_out=src/grpc -I ../../deps/concordium-grpc-api ../../deps/concordium-grpc-api/*/**/*.proto",
```

---

I also see that it has an option `output_javascript` which outputs js + .d.ts, which might be preferable, if we want to move the grpc out of the src folder.

---

I addressed the first comments, but I am having issues getting the `output_javascript` thing to work. I see your point however